be/src/agent/agent_server.cpp

```diff
@@ -602,18 +612,28 @@ void AgentServer::Impl::publish_cluster_state(TAgentResult& t_agent_result, cons
 void AgentServer::Impl::update_max_thread_by_type(int type, int new_val) {
     Status st;
     switch (type) {
-    case TTaskType::CLONE:
-        st = _thread_pool_clone->update_max_threads(new_val);
+    case TTaskType::UPLOAD:
+        st = _thread_pool_upload->update_max_threads(calc_real_num_threads(new_val));
+        break;
+    case TTaskType::DOWNLOAD:
+        st = _thread_pool_download->update_max_threads(calc_real_num_threads(new_val));
         break;
     case TTaskType::REMOTE_SNAPSHOT:
-        st = _thread_pool_remote_snapshot->update_max_threads(calc_max_replication_threads(new_val));
+        st = _thread_pool_remote_snapshot->update_max_threads(
+                calc_real_num_threads(new_val, REPLICATION_CPU_CORES_MULTIPLIER));
         break;
     case TTaskType::REPLICATE_SNAPSHOT:
-        st = _thread_pool_replicate_snapshot->update_max_threads(calc_max_replication_threads(new_val));
+        st = _thread_pool_replicate_snapshot->update_max_threads(
+                calc_real_num_threads(new_val, REPLICATION_CPU_CORES_MULTIPLIER));
         break;
-    default:
+    default: {
+        ThreadPool* thread_pool = get_thread_pool(type);
+        if (thread_pool) {
```

Comment:
maybe need to log warning info if the thread pool not exist?

Reply:
done